sdks/python/apache_beam/ml/transforms/embeddings/open_ai.py

```diff
@@ -0,0 +1,223 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# OpenAI Python SDK is required for this module.
+# Install using: pip install openai
+
+import logging
+import time
+from collections.abc import Iterable
+from collections.abc import Sequence
+from typing import Any, Optional, TypeVar, Union, Tuple, cast
```

Comment:
Prefer to use the built-in `tuple` over the typing types - see https://peps.python.org/pep-0585/